### PR TITLE
normalize ActiveSupport::Notification exceptions into error / error_d…

### PR DESCRIPTION
**Pull request by leviwilson**
## Which problem is this PR solving?
There was a discussion in the [`#sdk-ruby`](https://honeycombpollinators.slack.com/archives/CJR134U2F/p1630081866001300)  channel regarding `ActiveSupport::Notification` payloads normalizing the `payload[:exception]` similarly to the way that other integrations in the beeline by setting `"error"` and `"error_detail"` on the span.

Making this consistent will assist in querying traces that might contain exceptions.

## Short description of the changes
* handle when a `payload[:exception]` exists by adding:
  * `error` - the exception class
  * `error_detail` - the exception message

One decision that I was not clear about was what to do about the existing information on the payload. This means that if you have an event such at `"muh.custom.thing"` and an error occurrs, you will have this:

```json
{
  "muh.custom.thing.exception": ["StandardError", "I tried, but I have failed"],
  "error": "StandardError",
  "error_detail": "I tried, but I have failed"
}
```

My thoughts regarding this were to leave it untouched, so as to not break any expectations.

### CODE REVIEW

**Review by martin308 — COMMENTED** [review]
Will leave the approval up to @robbkidd. This looks good to me though as an idea. 

My only suggestion here would be to add some negative test cases around different values being in the `exception` key. It looks like this code would handle them but would be nice to codify that behaviour.

File: lib/honeycomb/integrations/active_support.rb
Comment (robbkidd) [inline lib/honeycomb/integrations/active_support.rb:54]: Aww yiss, array expansion. I think, though, that since you're grabbing the key off `payload` this ought to happen outside the `payload.each do … end` loop. The end result is correct, but this is extra work inside the loop.

File: spec/honeycomb/integrations/active_support_spec.rb
Comment (robbkidd) [inline spec/honeycomb/integrations/active_support_spec.rb:176]: This was a dope test. 👍🏻 

It helped me get my brain around adding a whole new case with nothing very custom for the default handler. I'd like to push that test case to this PR's branch, if you are OK with that and have the maintainers can update this PR enabled.

**Review by robbkidd — CHANGES_REQUESTED** [review]
A couple of comments inline with the code.

* The assignment and `add_field`ing needs to move out of the `payload.each` loop.
* Consider reaching for the `exception_object` instead of `exception` on `payload`.

File: lib/honeycomb/integrations/active_support.rb
Comment (leviwilson) [reply lib/honeycomb/integrations/active_support.rb:54]: Derp...not sure why I didn't see that :+1:

File: spec/honeycomb/integrations/active_support_spec.rb
Comment (leviwilson) [reply spec/honeycomb/integrations/active_support_spec.rb:176]: @robbkidd go for it (and I did choose that option)

**Review by robbkidd — APPROVED** [review]
Okay. Let's ride.

**Comment by robbkidd** [comment]
Thank you for your contribution, @leviwilson!

**Pull request merged**
